karate-tests/src/test/java/mocks/sandbox/patch-patient.js

```diff
@@ -89,6 +89,16 @@ function patchPatient (originalPatient, request) {
   if (request.body.patches.length === 0) {
     return setInvalidUpdateError(request, NO_PATCHES_PROVIDED)
   }
+  const updateErrors = []
+
+  if (request.body.patches.some(patch => patch.path.startsWith('/address'))) {
+    // Address is not whole object
+    if (!request.body.patches.some(patch => patch.path === '/address/0')) {
+      if (!request.body.patches.some(patch => patch.path === '/address/0/id')) {
+        updateErrors.push('Invalid update with error - no id or url found for path with root /address/0')
+      }
+    }
```

Comment:
I'm struggling to follow what's going on here. Is this saying that /address/0 and /address/0/id are the only valid values, and then anything outside of this for address gives this message?

Reply:
Just added a simple logic to support specific fields on a list item and whole object on the list item tests. if update is for specific fields then patch request must have a id  `/address/0/id` and it's not mandatory for whole object update `/address/0`